<?xml version="1.0"?>
<launch>
  <!-- Resolution in meter for the map when starting with an empty map.
        Otherwise the loaded file's resolution is used. -->
  <arg name="uav_id"              default="0"/>
  <arg name="uav_name"              default="uav$(arg uav_id)"/>
  <arg name="uav_takeoff_z"     default="2.0"/>

  <node pkg="mipp_packages" type="uav_server_node" name="uav_server_node" output="screen">
    <param name="uav_id"        value="$(arg uav_id)"/>
    <param name="uav_takeoff_z"        value="$(arg uav_takeoff_z)"/>

    <!-- Remap topics -->
    <!-- Publiches topics -->
    <remap from="uav_server/mavros_setpoint" to="mavros/setpoint_position/local"/>
    <!-- Subscribed topics -->
    <remap from="uav_server/local_goal" to="uav_server/local_goal"/>
    <remap from="uav_server/mavros_state" to="mavros/state"/>
    <remap from="uav_server/ground_truth_uav" to="/gazebo/ground_truth_$(arg uav_name)"/>
    <!-- Clients -->
    <remap from="uav_server/arm" to="mavros/cmd/arming"/>
    <remap from="uav_server/set_mode" to="mavros/set_mode"/>
  </node>
</launch>

